.github/workflows/yocto-build.yml

```diff
@@ -408,4 +401,179 @@ jobs:
             ${{ env.DEPLOY_PATH }}/*.*
             ${{ env.DEPLOY_PATH }}/VERSION*
             ${{ env.DEPLOY_PATH }}/image/balena.img
-     
\ No newline at end of file
+
+
+  ##############################
+  # Leviathan Test
+  ##############################
+  test:
+    runs-on: actuated-8cpu-8gb
+    needs: build
+    # runs-on: ${{ fromJSON(inputs.runs-on) }}
+    environment: ${{ inputs.environment }}
+
+    defaults:
+      run:
+        working-directory: .
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - DEVICE_TYPE: ${{ needs.build.outputs.device_slug }}
+          WORKER_TYPE: ${{ needs.build.outputs.worker_type }}
+          ENVIRONMENT: ${{ inputs.environment }}
+          TEST_SUITE: os
+        - DEVICE_TYPE: ${{ needs.build.outputs.device_slug }}
+          WORKER_TYPE: qemu
+          ENVIRONMENT: ${{ inputs.environment }}
+          TEST_SUITE: hup
+        - DEVICE_TYPE: ${{ needs.build.outputs.device_slug }}
+          WORKER_TYPE: qemu
+          ENVIRONMENT: ${{ inputs.environment }}
+          TEST_SUITE: cloud
+    steps:
+      - name: Reject unapproved external contributions
```

Comment:
Since this job depends on the `build` job, we really only need to add this rejection step to the build.